examples/models/llama3_2_vision/model.py

```diff
@@ -0,0 +1,174 @@
+# Copyright (c) Meta Platforms, Inc. and affiliates.
+# All rights reserved.
+#
+# This source code is licensed under the BSD-style license found in the
+# LICENSE file in the root directory of this source tree.
+
+# pyre-unsafe
+
+import json
+from typing import Any, Dict
+
+import torch
+from executorch.examples.models.checkpoint import (
+    get_checkpoint_dtype,
+    get_default_model_resource_dir,
+)
+
+from executorch.examples.models.model_base import EagerModelBase
+from torchtune.models.llama3_2_vision._component_builders import llama3_2_vision_decoder
+from torchtune.models.llama3_2_vision._convert_weights import llama3_vision_meta_to_tune
+
+
+def to_decoder_checkpoint(checkpoint: Dict[str, Any]) -> Dict[str, Any]:
```

Comment:
It seems the model is text decoder? Do you think it makes sense to create a subdirectory for text_decoder similar to vision encoder here:  https://github.com/pytorch/executorch/blob/main/examples/models/llama3_2_vision/vision_encoder/model.py

Reply:
I think that makes sense, only thing is it would muddle up some the eager model factory/base code since it’s opinionated on the file structure s.t. the file path to the model file is always example/models/<model name>/model.py. I can follow up with a PR to change this and move it into a subdue

Reply:
You can add this class to `examples/models/llama3_2_vision/__init__.py`